docs/Architecture Decision Record/009-git-workflow.md

```diff
@@ -0,0 +1,97 @@
+
+Throughout the project all vendor development work has been done in the vendor's forked repository, while pull requests from the government and even some documentation pull requests from the vendor were made directly to the government repository.
+
+This has created a situation where the vendor has needed to continuously rebase with the government repository to make sure the vendor's repository was up to date. As a result, problems with the Git history have arisen that make it confusing for both the government and the vendor to track the history of the work.
+
+The proposed workflow below provides a remedy to these issues, as well as many others, as detailed in the Consequences section.
+
+## Decision
+
+A contributor to the TDP project would always use the following steps to propose changes to the repository. No merges directly into `HHS:main`
+
+![](images/TANF-Git-Workflow.png)
+
+1. Check out `raft-tdp-main` in `raft-tech/TANF-app` (or another branch if this work is dependent on a branch that hasn't been merged yet)
+2. If working locally, run `git pull`
+3. Check out new feature branch `git checkout -b <BRANCH NAME>`
+4. Update the repo with code, documentation, etc.
+5. If working locally, run `git push origin <BRANCH NAME>`
+6. Create [a draft pull request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests) from new branch to `raft-tdp-main`. The PR title should include WIP e.g. `WIP: Adding a feature` (If your work is dependent on another branch that has yet to be merged, issue the pull request against that branch)
+8. Implement, Test, Review work independently
+9. When finished: 
+    * Update the Pull Request Template
+    * Add in-line comment to the file changes to provide context for the proposed changes
+    * Ensure there are no merge conflicts 
+    * Ensure CI/CD pipelines are green
+    * Update the title to remove `WIP`, change the PR to [Ready for Review](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-stage-of-a-pull-request), assign label `raft-review`.
+    * If your PR is against a branch other than `raft-tdp-main` keep it as a draft PR and add the tag `blocked` until the other branch is merged. 
+13. Assign a reviewer: 
+    * For development work, assign **at least two** Raft developers with one of them being `carltonsmith` or `jtwillis92` (**but not both**).
+    * For research and design assign `shubhi-raft` and `jenewingpierce`
+    * For security controls assign `carltonsmith`
+    * For documentation submitted by the Government tag `lfrohlich` and `adpennington` and remove label `raft-review`
```

Comment:
I'm a bit confused. if the documentation is submitted by the government (which will soon just be me at @ADPennington ), why would we also be reviewers? Or does this presume the documentation is submitted by 18F and reviewed by OFA?

Reply:
This is just because every PR needs to be approved, and you don't need our approval for your documentation changes. The assumption is that if you put in a change, @ADPennington would review it, and if she put in a change, you would review it.

Reply:
ah ok. got it -- thanks!